source/analysis/REIXS/REIXSXESImageInterpolationAB.cpp

```diff
@@ -381,40 +382,44 @@ void REIXSXESImageInterpolationAB::computeCachedValues() const
 		double numY = (double)(sumRangeMaxY_ - sumRangeMinY_);
 
 
-		for(int i=0; i<iSize; ++i) {
-			double newVal = 0.0;
-			int contributingRows = 0;
-			if(i > sumRangeMinX_ && i < sumRangeMaxX_) {
-				double xVal = (double)i - originX;
-				for(int j=sumRangeMinY_; j<=sumRangeMaxY_; j++) { // loop through rows
-					if(rangeRound_ == 0.0) { //not ellipse
-						int sourceI = i + shiftValues1_.at(j);
-						if(sourceI < iSize && sourceI >= 0) {
-							newVal += image.at(sourceI*jSize + j);
-							contributingRows++;
-						}
-					}
-					else {
-
-						double yVal = (double)j - originY;
-						if((fabs(xVal) <= numX/2.0*(1.0 - rangeRound_)) || (fabs(yVal) <= numY/2.0*(1.0 - rangeRound_)) || ((((xVal-(1-rangeRound_)*numX/2.0)/(rangeRound_*numX/2.0))*((xVal-(1-rangeRound_)*numX/2.0)/(rangeRound_*numX/2.0))+((yVal-(1-rangeRound_)*numY/2.0)/(rangeRound_*numY/2.0))*((yVal-(1-rangeRound_)*numY/2.0)/(rangeRound_*numY/2.0))) < 1)) { //within ellipse
+		for(int b = 0; b < iSize / binningLevel_; b++) {
+			cachedValues_[b] = 0;
+			for(int temp = 0; temp < binningLevel_; temp++ ) {
+				int i = b * binningLevel_ + temp;
+				double newVal = 0.0;
+				int contributingRows = 0;
+				if(i > sumRangeMinX_ && i < sumRangeMaxX_) {
+					double xVal = (double)i - originX;
+					for(int j=sumRangeMinY_; j<=sumRangeMaxY_; j++) { // loop through rows
+						if(rangeRound_ == 0.0) { //not ellipse
 							int sourceI = i + shiftValues1_.at(j);
 							if(sourceI < iSize && sourceI >= 0) {
 								newVal += image.at(sourceI*jSize + j);
 								contributingRows++;
 							}
 						}
+						else {
+
+							double yVal = (double)j - originY;
+							if((fabs(xVal) <= numX/2.0*(1.0 - rangeRound_)) || (fabs(yVal) <= numY/2.0*(1.0 - rangeRound_)) || ((((xVal-(1-rangeRound_)*numX/2.0)/(rangeRound_*numX/2.0))*((xVal-(1-rangeRound_)*numX/2.0)/(rangeRound_*numX/2.0))+((yVal-(1-rangeRound_)*numY/2.0)/(rangeRound_*numY/2.0))*((yVal-(1-rangeRound_)*numY/2.0)/(rangeRound_*numY/2.0))) < 1)) { //within ellipse

```

Comment:
Considering how long that is, it might be better to make a helper function like `isEllipse()` and then be really explicit about the tests.